### PR TITLE
Possibility of enable 5Ghz through java reflection

### PR DESCRIPTION
**Pull request by forlayo**
Added the possibility of enable 5Ghz through java reflection ( Maybe it will only works on Android 6 only ).

### CODE REVIEW

**Comment by mvdan** [comment]
Did you try this? Does it work? (can't try it myself because this lib is currently broken on 7.1, which is what I run)

**Comment by mvdan** [comment]
Also, add "Fixes #13" in the body of the commit message so that the issue will be closed when this is merged.

**Comment by forlayo** [comment]
Yep, it works; I've been sucessfully used it on my Nexus 6p and Nexus 5x devices. Can I modify the commit here? I'm leaving the office right now and I haven't access to the git console right now :(

**Comment by mvdan** [comment]
I don't think the web UI lets you modify commits - but that's okay, I'm not in a rush to merge this. I'll review the code now.

File: library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java [hunk 1]
```diff
@@ -436,4 +437,57 @@ public void run() {
...
 	}
+
+	/**
+	 * Checks if 5Ghz band is available to create an accesspoint.
+	 * @return
+	 */
+	public static boolean is5GhzBandAvailable(){
+		WifiConfiguration wifiConfiguration = new WifiConfiguration();
+		return setField(wifiConfiguration,"apBand",1);
+	}
```
Comment (mvdan) [inline library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java:448]: This should only check if the field exists, not change it.

File: library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java [hunk 2]
```diff
@@ -436,4 +437,57 @@ public void run() {
...
+
+	/**
+	 * Sets a field value on a given object
+	 *
+	 * @param targetObject the object to set the field value on
+	 * @param fieldName    exact name of the field
+	 * @param fieldValue   value to set on the field
+	 * @return true if the value was successfully set, false otherwise
+	 */
+	public static boolean setField(Object targetObject, String fieldName, Object fieldValue) {
```
Comment (mvdan) [inline library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java:458]: private

File: library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java [hunk 3]
```diff
@@ -436,4 +437,57 @@ public void run() {
 		}.start();
 		return clients;
 	}
+
+	/**
+	 * Checks if 5Ghz band is available to create an accesspoint.
+	 * @return
+	 */
```
Comment (mvdan) [inline library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java:444]: Use shorter method docs, like in the rest of the code:

```
// Checks if 5Ghz band is available to create an accesspoint.
```

Same for the rest of the code you added.

File: library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java [hunk 4]
```diff
@@ -436,4 +437,57 @@ public void run() {
...
+	}
+
+	/**
+	 * Tries to enable 5Ghz band for accesspoint in provided configuration.
+	 * @param configuration
+	 * @return
+	 */
+	public static boolean enable5GhzBand(WifiConfiguration configuration){
+		return setField(configuration, "apBand", 1);
+	}
```
Comment (mvdan) [inline library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java:492]: What about switching to 2GHz?

File: library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java [hunk 5]
```diff
@@ -436,4 +437,57 @@ public void run() {
...
+			field = null;
+		}
+		Class superClass = targetObject.getClass().getSuperclass();
+		while (field == null && superClass != null) {
+			try {
+				field = superClass.getDeclaredField(fieldName);
+			} catch (NoSuchFieldException e) {
+				superClass = superClass.getSuperclass();
+			}
+		}
```
Comment (mvdan) [inline library/src/main/java/cc/mvdan/accesspoint/WifiApControl.java:472]: Why is this `while` needed? (just curious)

**Review by mvdan — COMMENTED** [review]
I would change the API to not be centralized on 5GHz. Done like this, once you switch to 5GHz there is no way to go back to 2.4GHz.

Perhaps do something like:

```
public enum Band {
    2GHz, 5GHz
}

public static boolean setBand(Band band) { ...
```

**Comment by mvdan** [comment]
And thanks for contributing! :)

**Review by mvdan — CHANGES_REQUESTED** [review]
I also just realised that there is no way to get the current band. As it is now, this change has limited use and it exposes too much.

**Comment by mvdan** [comment]
Bump @forlayo

**Comment by mvdan** [comment]
Ping @forlayo - happy to get the feature merged, but not in the current state. Do you have time to fix the pull request?

**Comment by mvdan** [comment]
Still no answer - closing.

**Pull request closed without merging**